reviewing/dependencies-hierarchy/src/buildDependenciesHierarchy.ts

```diff
@@ -107,23 +108,28 @@ async function dependenciesHierarchyForPackage (
 
   const getChildrenTree = getTree.bind(null, {
     currentPackages: currentLockfile.packages ?? {},
+    importers: currentLockfile.importers,
     includeOptionalDependencies: opts.include.optionalDependencies,
     lockfileDir: opts.lockfileDir,
+    rewriteLinkVersionDir: projectPath,
```

Comment:
Note that the `link:...` paths won't be relative to the current working directory when running the list command with the `--recursive` flag: `pnpm ls -r`

In that case the `projectPath` isn't the same as `process.cwd()`.

This is the case before the changes in this PR. I followed the existing convention of printing relative to `projectPath`.

https://github.com/pnpm/pnpm/blob/37c818d1bd92c47050ee634f92d1f70002f0fb39/reviewing/dependencies-hierarchy/src/buildDependenciesHierarchy.ts#L167